praw/models/util.py

```diff
@@ -177,15 +181,9 @@ def stream_generator(
         if before_attribute is None:
             limit -= without_before_counter
             without_before_counter = (without_before_counter + 1) % 30
-        for item in reversed(
-            list(
-                function(
-                    limit=limit,
-                    params={"before": before_attribute},
-                    **function_kwargs
-                )
-            )
-        ):
+        if not exclude_before:
+            function_kwargs["params"] = {"before": before_attribute}
```

Comment:
Will `function_kwargs["params"]` ever exist as passed into the function? The previous code seems to assume it will not but I want to double-check.

Reply:
It is how the `before` is passed into `func` listing generator.

Reply:
I added the conditional before_attribute because `modal_conversations` doesn't accept the `params` arg.

Reply:
This was suggested by @bboe when I chatted with him on gitter.

Reply:
Yes, I see that that is what you use it for. What I was wondering is whether those parameters will ever be supplied by a user, in which case your code would override them.

Reply:
I don't believe so. Only time I have used them is to specify before, which the streams automatically set with the exception of `modmail_conversations` stream.

Reply:
Cool, thanks for the response.